R/vctr.R

```diff
@@ -563,7 +567,8 @@ levels.vctrs_vctr <- function(x) {
 
 #' @export
 `is.na<-.vctrs_vctr` <- function(x, value) {
-  stop_unsupported(x, "is.na<-")
+  vec_slice(x, value) <- vec_na(x)
```

Comment:
I think this needs to assert that `value` is a logical vector and is size-compatible with `x`

Reply:
This is done in `vec_slice<-()` . Integers and characters are also supported in base:

``` r
x <- 1:3
is.na(x) <- 2
x
#> [1]  1 NA  3

x <- c(a = 1, b = 2, c = 3)
is.na(x) <- "b"
x
#>  a  b  c 
#>  1 NA  3
```

<sup>Created on 2019-01-29 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

Reply:
That feels overly broad to me; I think it makes more sense to only accept logical vectors.

Reply:
I see the following invariant:

```r
identical(`is.na<-`(x, i), `[<-`(x, i, NA))
```

It makes perfect sense to me to just pass through to `vec_slice<-()` , given that special-casing just adds complexity here.

Reply:
`vec_assert(values, logical())` adds little complexity and I don't think that a broad interpretation of `values` is useful here.